typespec-tests/tsp/patch.tsp

```diff
@@ -23,16 +26,66 @@ model Resource {
   name: string;
 
   description?: string;
-  type: string;
+  map: Record<InnerModel>;
+  longValue?: int64;
+  intValue?: int32;
+  enumValue?: "a" | "b" | "c";
+
+  @projectedName("json", "wireNameForInnerModelProperty")
+  innerModelProperty?: InnerModel;
+}
+
+model InnerModel {
+  name: string;
```

Comment:
Add an optional property to InnerModel.

Then you are able to test `{"map": {"key": {"optional": null}}}`